crates/polars-sql/src/sql_expr.rs

```diff
@@ -448,7 +463,11 @@ impl SqlExprVisitor<'_> {
             None => None,
             _ => return self.err(&expr),
         };
-
+        match trim_characters {
+            // TODO: allow compact snowflake/bigquery syntax?
+            Some(_) => return Err(polars_err!(ComputeError: "unsupported TRIM syntax")),
+            None => (),
+        };
```

Comment:
```suggestion
        if trim_characters.is_some() {
            // TODO: allow compact snowflake/bigquery syntax?
            return Err(polars_err!(ComputeError: "unsupported TRIM syntax")),
        };
```

Reply:
we could probably also move this to the top of the function to avoid any unnecessary work.

Reply:
Was thinking we could make it work later too; a fairly low-stakes/convenient addition 🤔